api/tests/Feature/PoolCandidateUpdateTest.php

```diff
@@ -502,12 +504,66 @@ public function testPlaceCandidateMutation(): void
         assertSame($response['placedDepartment']['id'], $department->id);
     }
 
+    // placing candidates sets suspended_at automatically or keeps/makes it null appropriately
+    public function testPlaceCandidateMutationSuspensionLogic(): void
+    {
+        $department = Department::factory()->create();
+        $this->poolCandidate->pool_candidate_status = PoolCandidateStatus::QUALIFIED_AVAILABLE->name;
+        $this->poolCandidate->placed_at = null;
+        $this->poolCandidate->save();
+        $placedDoNotSuspend = [
+            PlacementType::PLACED_TENTATIVE->name,
+            PlacementType::PLACED_CASUAL->name,
+        ];
+        $placedDoSuspend = [
+            PlacementType::PLACED_TERM->name,
+            PlacementType::PLACED_INDETERMINATE->name,
+        ];
+
+        foreach ($placedDoNotSuspend as $value) {
```

Comment:
Can we be more specific?
```suggestion
        foreach ($placedDoNotSuspend as $placementType) {
```

Reply:
https://github.com/GCTC-NTGC/gc-digital-talent/pull/12330/commits/9af9134ea554e83f7db886038b86be858c05f194